common/lib/capa/capa/responsetypes.py

```diff
@@ -1083,16 +1255,80 @@ def late_transforms(self, problem):
         self.do_shuffle(self.xml, problem)
         self.do_answer_pool(self.xml, problem)
 
-    def get_score(self, student_answers):
+    def grade_via_points(self, **kwargs):
```

Comment:
I'm having a hard time reviewing this because it looks like you've changed things from your previous pull request. For example, this function was in a different place, with a different docstring and implementation, in https://github.com/edx/edx-platform/pull/8724/files#diff-38722d7e328c77e747d36e85e7ab2c0fR1590

I'm very concerned if you've made additional changes. What would be much better would be if the pr was exactly thes ame in every respect, except with a few more lines removed. I don't like changing docstrings or changing implementations.